unbxdAutosuggest.js

```diff
@@ -1803,10 +1803,19 @@ var unbxdAutoSuggestFunction = function ($, Handlebars, params) {
 				+ '{{/if}}';
 		}
 		, prepareHTML: function () {
-			var html = '<ul class="unbxd-as-maincontent unbxd-as-suggestions-overall">',
-				self = this,
-				mainlen = 0,
-				sidelen = 0;
+			var html = '<ul class="unbxd-as-maincontent unbxd-as-suggestions-overall">';
+
+			if (this.options.isMobile) {
+				if (this.options.isMobile()) {
+					html = '<ul class="unbxd-as-maincontent unbxd-as-suggestions-overall mobile">';
+				}
+			} else if (isMobile.any()) {
+				html = '<ul class="unbxd-as-maincontent unbxd-as-suggestions-overall mobile">';
+			}
```

Comment:
can it be simplified as

```
var isMobileFn = this.options.isMobile;

if ((typeof(isMobileFn) === "function" && isMobileFn()) || isMobile.any()) {
	html = '<ul class="unbxd-as-maincontent unbxd-as-suggestions-overall mobile">';
}
```

Reply:
As the same HTML code is repeated twice, any change will have to be done at two places, hence a maintenance overhead

Reply:
If Customer defined isMobile returns to false, it will still check the library isMobile and add the mobile classes and HMTL which is incorrect. Hence the latter check in else if.

Reply:
Got it. You can save it in variable to reuse inside both the if cases then.

Reply:
Done